File: enode/src/main/java/org/enodeframework/eventing/IPublishedVersionStore.java
package org.enodeframework.eventing;

import java.util.concurrent.CompletableFuture;

public interface IPublishedVersionStore {
    /**
     * Update the published version for the given aggregate.
     */
    CompletableFuture<Integer> updatePublishedVersionAsync(String processorName, String aggregateRootTypeName, String aggregateRootId, int publishedVersion);

    /**
     * Get the current published version for the given aggregate.
     */
    CompletableFuture<Integer> getPublishedVersionAsync(String processorName, String aggregateRootTypeName, String aggregateRootId);
}

File: enode/src/main/java/org/enodeframework/eventing/IPublishedVersionStore.kt
package org.enodeframework.eventing

import java.util.concurrent.CompletableFuture

interface IPublishedVersionStore {
    /**
     * Update the published version for the given aggregate.
     */
    fun updatePublishedVersionAsync(processorName: String, aggregateRootTypeName: String, aggregateRootId: String, publishedVersion: Int): CompletableFuture<Int>

    /**
     * Get the current published version for the given aggregate.
     */
    fun getPublishedVersionAsync(processorName: String, aggregateRootTypeName: String, aggregateRootId: String): CompletableFuture<Int>
}